frontend/lib/framework/error-classes-list.cpp

```diff
@@ -537,4 +538,189 @@ void ErrorProcDefExplicitAnonFormal::write(ErrorWriterBase& wr) const {
   wr.code(fn, {formal});
 }
 
+/* parser errors */
+
+void ErrorParsing::write(ErrorWriterBase& wr) const {
+  auto loc = std::get<const Location>(info);
+  auto errorMessage = std::get<std::string>(info);
+  assert(errorMessage.back() == '.' &&
+         "expected a period at the end of ErrorParsing message");
+  wr.heading(kind_, type_, loc, errorMessage);
+}
+
+void ErrorCannotAttachPragmas::write(ErrorWriterBase& wr) const {
+  auto loc = std::get<const Location>(info);
+  auto stmt = std::get<const uast::AstNode*>(info);
+  wr.heading(kind_, type_, loc, "cannot attach 'pragma's to this statement.");
+  if (stmt) {
+    wr.code(stmt);
+  }
+  wr.message(
+      "Only declarations, such as variable declarations, can have 'pragma's "
+      "attached to them.");
+}
+
+void ErrorInvalidIndexExpr::write(ErrorWriterBase& wr) const {
+  auto loc = std::get<const Location>(info);
+  wr.heading(kind_, type_, loc,
+             "only identifiers or tuples are allowed as index expressions.");
```

Comment:
maybe "tuples of identifiers" instead of just "tuples"? `(x,1)` is a tuple, but isn't a valid index expression.